extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -894,18 +894,32 @@ class LifecycleTask extends BackbeatTask {
                 return process.nextTick(cb);
             }
 
-            return async.waterfall([
-                next => this._getRules(bucketData, lcRules, obj, log, next),
-                (applicableRules, next) => {
-                    if (versioningStatus === 'Enabled'
-                    || versioningStatus === 'Suspended') {
-                        return this._compareVersion(bucketData, obj, contents,
-                            applicableRules, versioningStatus, log, next);
-                    }
-                    return this._compareObject(bucketData, obj, applicableRules, log,
-                        next);
+            // We don't want to retry the _whole_ list if only a single entry fails,
+            // so we possibly retry each individual entry here, and ignore errors.
```

Comment:
Do we even need to retry?

Reply:
I think we should retry, at least until lifecycle optimization is used everywhere and we can run lifecycle more often : otherwise a minor retriable error could delay lifecycle for 12h.

That said, indeed retry has delay which we need to consider, esp. w/ respect to being a kafka consumer.

Reply:
actually, looking at the code of `_applyTransitionRule()`, we already skip the errors from `s3:_getObjectMD`... though some errors can still go through, e.g. when  calling in `s3:listObjectVersions` ( `_checkAndApplyEODMRule()`) or `s3:headObject` in non-versionned bucket...